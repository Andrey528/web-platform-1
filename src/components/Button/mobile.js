import Button from './view/mobile/Button';
export default Button;