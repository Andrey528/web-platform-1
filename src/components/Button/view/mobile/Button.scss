.button {
  width: 100%;
  min-height: 1.875rem;
  display: flex;
  justify-content: center;
  align-items: center;
  color: white;
  text-transform: uppercase;
  cursor: pointer;
  box-sizing: border-box;
  padding: 0 0.5rem;
  border: none;
  font-weight: bold;
  white-space: nowrap;
  font-size: 0.75rem;
  border-radius: 0.3125rem;

  &:active {
    transform: translateY(2px);
  }
  user-select: none;
  text-align: center;

  &_colors {
    &_green {
      background: #1f4b9e;
    }
    &_gray {
      background: #9e1f1f;
    }
    &_additional-green {
      background: linear-gradient(180deg, #89BC37 0%, #008153 100%);
      box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.25), inset 0px 1px 0px rgba(255, 255, 255, 0.25);
      &:hover {
        background: linear-gradient(0deg, rgba(255, 255, 255, 0.2), rgba(255, 255, 255, 0.2)), linear-gradient(180deg, #5DDB52 0%, #01764C 100%);
      }
    }
    &_dark {
      background: #182533;
      color: #429BDB;
    }
    &_blue {
      background: #5288C1;
      &:hover {
        background: linear-gradient(0deg, rgba(255, 255, 255, 0.2), rgba(255, 255, 255, 0.2)), linear-gradient(180deg, #268BE9 0%, #0E5BB6 100%);
      }
    }
    &_dark-blue {
      background: #9e1f1f;
      color: #fff;
      &:hover {
        background: lighten(#9e1f1f, 15%)
      }
    }
  }

  &_sizable {
    &_default {
      height: 100%;
      text-decoration: none;
    }
    &_low {
      height: 2.5rem;
      text-decoration: none;
      color: #fff;
    }
    &_high {
      height: 3rem;
      text-decoration: none;
    }
  }

  &_disable {
    background: #1f4b9e;
  }
}