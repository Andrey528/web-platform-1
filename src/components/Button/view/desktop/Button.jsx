import React from 'react';
import block from 'bem-cn';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './Button.scss';

const Button = ({ text, callBack, disabled = false, size = 'default', color = 'green', type = 'button', link }) => {
  const b = block('button');
  return link ? (
    <Link
      className={b({ sizable: size }, { disable: disabled }, { colors: color })}
      to={link}
    >
      {text}
    </Link>) : (
      <button
        type={type}
        className={b({ sizable: size }, { disable: disabled }, { colors: color })}
        onClick={disabled ? f => f : callBack}
      >
        {text}
      </button>
  );
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  size: PropTypes.string,
  color: PropTypes.string,
  type: PropTypes.string,
  link: PropTypes.string,

  callBack: PropTypes.func,
};

export default Button;
