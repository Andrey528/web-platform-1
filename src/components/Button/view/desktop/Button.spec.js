import React from 'react';
import { shallow } from 'enzyme';
import block from 'bem-cn';

import Button from './Button';

describe('Button desktop tests', () => {
  const b = block('button');
  
  describe('Button with text', () => {
    const props = { text: 'Test text' };

    const button = shallow(<Button {...props} />);

    it('render properly', () => {
      expect(button).toMatchSnapshot();
    });

    it('render text', () => {
      expect(button.find('button').text()).toEqual(props.text);
    });
  });

  describe('Button with link', () => {
    const props = {
      text: 'Test link button',
      link: '/test-button',
    };

    const button = shallow(<Button {...props} />);

    it('render properly', () => {
      expect(button).toMatchSnapshot();
    });

    it('render link', () => {
      expect(button.find('Link')).toHaveLength(1);
    });

    it('render link with "to"', () => {
      expect(button.find('Link').props().to).toEqual(props.link);
    });
  });

  describe('Button with type', () => {
    const props = {
      text: 'Test',
    };

    it('Default type button', () => {
      const nextProps = {
        ...props,
      };

      const button = shallow(<Button {...nextProps} />);
      expect(button.find('button').props().type).toEqual('button');
    });

    it('Submit type button', () => {
      const nextProps = {
        ...props,
        type: 'submit',
      };

      const button = shallow(<Button {...nextProps} />);

      expect(button.find('button').props().type).toEqual(nextProps.type);
    });
  });

  describe('Button with modificators', () => {
    const props = {
      text: 'Test mods',
    };

    it('Have default modificators', () => {
      const className = b({ sizable: 'default' }, { disable: false }, { colors: 'green' });
      const button = shallow(<Button {...props} />);

      expect(button.find('button').props().className.toString()).toEqual(className.toString());
    });

    it('Different modificators', () => {
      const nextProps = {
        ...props,
        size: 'big',
        disabled: true,
        color: 'red',
      };
      const className = b({ sizable: nextProps.size }, { disable: true }, { colors: nextProps.color });
      const button = shallow(<Button {...nextProps} />);

      expect(button.find('button').props().className.toString()).toEqual(className.toString());
    });
  });

  describe('Test callBack work', () => {
    const mockCallBack = jest.fn();
    const props = {
      text: 'Test',
      callBack: mockCallBack,
    };

    it('onClick working', () => {
      const button = shallow(<Button {...props} />);

      button.find('button').simulate('click');

      expect(mockCallBack).toHaveBeenCalledTimes(1);
    });

    it('onClick multiply times', () => {
      const button = shallow(<Button {...props} />);

      button.find('button').simulate('click');
      button.find('button').simulate('click');
      button.find('button').simulate('click');

      expect(mockCallBack).toHaveBeenCalledTimes(3);
    });

    it('onClick disabled', () => {
      const nextProps = { ...props, disabled: true };
      const button = shallow(<Button {...nextProps} />);

      button.find('button').simulate('click');
      button.find('button').simulate('click');
      button.find('button').simulate('click');

      expect(mockCallBack).toHaveBeenCalledTimes(0);
    });
  });
});