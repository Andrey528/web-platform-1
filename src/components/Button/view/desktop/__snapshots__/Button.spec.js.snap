// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button desktop tests Button with link render properly 1`] = `
<Link
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  to="/test-button"
>
  Test link button
</Link>
`;

exports[`Button desktop tests Button with text render properly 1`] = `
<button
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  type="button"
>
  Test text
</button>
`;
